bin/que

```diff
@@ -97,25 +92,41 @@ end
 
 queue_name     = options.queue_name     || ENV['QUE_QUEUE'] || Que::Worker::DEFAULT_QUEUE
 wake_interval  = options.wake_interval  || ENV['QUE_WAKE_INTERVAL']&.to_f
-metrics_labels = options.metrics_labels || {}
 cursor_expiry  = options.cursor_expiry  || 0
 worker_count   = options.worker_count   || 1
 timeout        = options.timeout
 
-stop_workers = Que::Worker.start_workers(
+workers = Que::WorkerGroup.start(
   worker_count,
   queue: queue_name,
   wake_interval: wake_interval,
   lock_cursor_expiry: cursor_expiry,
-  metrics_labels: metrics_labels,
-  metrics_registry: Prometheus::Client.registry
 )
 
 if options.metrics_port
   Thread.new do
-    Que::Metrics.expose(Prometheus::Client.registry, port: options.metrics_port)
+    Que.logger&.info(
+      event: "serving_metrics",
+      msg: "Serving /metrics endpoint",
+      port: options.metrics_port,
+    )
+
+    app = Rack::Builder.new do
+      use Que::Middleware::WorkerCollector, workers: workers
+      use Prometheus::Middleware::Exporter
+
+      run lambda { |env| [200, {}, ["healthy"]] }
+    end
+
+    Rack::Handler::WEBrick.run(
```

Comment:
It's not a huge deal, but webrick is a generally considered a poor choice for a production web server, so this doesn't set a good example to others. It might be nicer to use Thin instead.

Reply:
Yeah this is a good point, now we're actually bundling it there's no point avoiding the requirement now.

Reply:
On second thoughts (and after trying to use thin) I don't reckon we should do this. Servers like thin think about the world in a very different way from webrick, such as "I should control signals" and "lets boot eventmachine to handle incoming requests" which can really interfere with how Que behaves.

I reckon we don't suffer from using webrick right now, as this endpoint will never receive many requests, and webrick is better suited to boot an in-memory server that keeps to itself.

Wdyt?

Reply:
I buy that - we definitely don't want anything else handling signals or weird things could happen